File: py/prob_26.py
class Solution:
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        
        carret = 0
        for i in range(1, len(nums)):
            if nums[carret] != nums[i]:
                carret += 1
                nums[carret] = nums[i]
        
        return carret + 1

File: cpp/prob_26.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (!nums.size()) {
            return 0;
        }
        
        int carret = 0;
        for (int i=1; i < nums.size(); i++) {
            if (nums[carret] != nums[i]) {
                carret++;
                nums[carret] = nums[i];
            }
        }
        return carret + 1;
    }
};